pandas/_libs/tslib.pyx

```diff
@@ -666,6 +668,16 @@ cdef _array_to_datetime_object(
 
         cnp.PyArray_MultiIter_NEXT(mi)
 
+    if len(unique_timezones) > 1:
+        warnings.warn(
+            "In a future version of pandas, parsing datetimes with mixed time "
+            "zones will raise a warning unless `utc=True`. "
```

Comment:
The message here says that this will raise a _warning_ in the future. But is that indeed the intent, or should that be "error" instead? 
(my understanding of the discussion was that it would error in the future)

Reply:
you're right, thanks - @natmokval fancy addressing this in a separate PR?

Reply:
sure, I replaced "warning" with "error" in warning message and made a new PR